homeassistant/components/switch/xiaomi_miio.py

```diff
@@ -124,14 +128,18 @@
         except DeviceException:
             raise PlatformNotReady
 
-    if model in ['chuangmi.plug.v1']:
-        from miio import PlugV1
-        plug = PlugV1(host, token)
+    if model in ['chuangmi.plug.v1', 'chuangmi.plug.v3']:
+        if model == MODEL_PLUG_V3:
+            from miio import PlugV3
+            plug = PlugV3(host, token)
+        else:
+            from miio import PlugV1
+            plug = PlugV1(host, token)
```

Comment:
With python-miio maintainer hat on, should we try to merge PlugV1 and PlugV3 into one class (and allow specifying the type with an enum)?

Reply:
Let's merge both devices into one class! :+1:

Reply:
Ok, great! This depends on https://github.com/rytilahti/python-miio/pull/270 and a new version of python-miio then, I'll mark it as a WIP.